library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -124,6 +124,34 @@
   protected transient long skipCount = 0;
   private transient OperatorContext context;
 
+  /**
+   * The time between sending batch tuples. Default value is 1 hour. A batch
+   * interval could consist of multiple scan intervals
+   */
+  private long batchIntervalMillis = 60 * 60 * 1000; //1 hour
```

Comment:
The point is that typically the batch boundary is determined based on the input, not on system clock. What is lacking here is the necessary abstraction for triggering a batch tuple.